app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Save.php

```diff
@@ -127,8 +143,10 @@ public function execute()
             $attributeId = $this->getRequest()->getParam('attribute_id');
             $attributeCode = $this->getRequest()->getParam('attribute_code')
                 ?: $this->generateCode($this->getRequest()->getParam('frontend_label')[0]);
-            if (strlen($attributeCode) > 0) {
-                $validatorAttrCode = new \Zend_Validate_Regex(['pattern' => '/^[a-z\x{600}-\x{6FF}][a-z\x{600}-\x{6FF}_0-9]{0,30}$/u']);
+            if (!$attributeId && strlen($attributeCode) > 0) {
```

Comment:
I believe it would be safer to validate attribute code even for existing attributes like it was before.

Reply:
Basically here is the appropriate logic change as to me: https://github.com/magento/magento2/issues/6770#issuecomment-255696090

We just should not obtain attribute code from request or generate it for already existing attributes.